apps/meteor/app/threads/server/methods/getThreadMessages.js

```diff
@@ -28,18 +29,24 @@ Meteor.methods({
 
 		const user = Meteor.user();
 		const room = Rooms.findOneById(thread.rid);
+		callbacks.run('beforeReadMessages', thread.rid, user._id);
```

Comment:
this should be moved to right before the call to `readThread(..)`, after all checks are done

Reply:
tks, done!